File: DSA/Shubham_590016506/Day_5_Question_3.c
#include <stdio.h>
#include <stdbool.h>


bool isPalindrome(int* nums, int size) {
    for (int i = 0; i < size / 2; i++) {
        if (nums[i] != nums[size - 1 - i]) {
            return false; 
        }
    }
    return true;  
}

int main() {
    int nums[] = {1, 2, 3, 2, 1};
    int size = sizeof(nums) / sizeof(nums[0]);
    printf("Is palindrome: %s\n", isPalindrome(nums, size) ? "true" : "false"); // Output: true
    return 0;
}

File: DSA/pranavdixit_590014418/Day5Question3.py
def is_palindrome(nums):
    return nums == nums[::-1]

# Example usage:
print(is_palindrome([1, 2, 3, 2, 1])) 
print(is_palindrome([1, 2, 3, 4]))    
print(is_palindrome([5, 5, 5, 5]))    
